usr/sbin/p11sak/p11sak.c

```diff
@@ -722,86 +800,53 @@ static CK_RV key_pair_gen_mech(p11sak_kt kt, CK_MECHANISM *pmech)
 
     return CKR_OK;
 }
+
 /**
- * Set default asymmetric key attributes.
+ * returns 1 if the given attribute is not applicable for the
+ * given key type, 0 otherwise.
  */
-static CK_RV set_battr(const char *attr_string, CK_ATTRIBUTE *pubattr,
-                       CK_ULONG *pubcount, CK_ATTRIBUTE *prvattr,
-                       CK_ULONG *prvcount)
+static CK_BBOOL attr_na(const CK_ULONG attr_type, p11sak_kt ktype)
```

Comment:
Function `attr_na()` is OK as is for now, but once we allow key creation with config-defined attributes, we need to add code to allow such config-defined attributes for key objects as well.

Reply:
for future work...